### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by djo1975**
**In this project, I used migration files to build my schema.**

1.     Created a new app.
2.     Setuped Postgres database set up.
3.     Created and ran the necessary migration files.
4.     Table and column names match the ERD diagram.
5.     Foreign keys included.
6.     In the posts table the column holding the foreign key to the user's table named author_id
7.     All columns that are foreign keys have a corresponding index.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
```
Comment (tresorsawasawa) [inline LICENSE:2]: - [ ] As required for the project and according to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) provided to you, kindly make sure the migration file for creating users table has `columns` matching the one from the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) and with correct `data_type`. Here is how your table should look like:
```ruby
create_table :users do |t|
    t.string :name
    t.string :photo 
    t.text :bio
    t.integer :posts_counter

    t.timestamps 
  end
```

Comment (tresorsawasawa) [inline LICENSE:2]: - [ ] To match the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)[ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) and the project requirements, kindly ensure that your `posts` table has column for `foreign_key` with corresponding index. I suggest you add that column and index in separate migration file. **When creating the reference column you should ensure that your `Post` belongs to an `:author`**. To help you just refer this **[article](https://stackoverflow.com/questions/27809342/rails-migration-add-reference-to-table-but-different-column-name-for-foreign-ke/42214988)**

Comment (tresorsawasawa) [inline LICENSE:2]: - [ ] To match the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) and the project requirements, kindly ensure that your `comments` table has column for `foreign_key` with corresponding index(for posts and users table). I suggest you add those columns and index in separate migration file. **When creating the reference column you should ensure that your `Comment` belongs to an `:author`** and not to `user`. To help you just refer this **[article](https://stackoverflow.com/questions/27809342/rails-migration-add-reference-to-table-but-different-column-name-for-foreign-ke/42214988)**

Comment (tresorsawasawa) [inline LICENSE:2]: - [ ] To match the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) and the project requirements, kindly ensure that your `likes` table has column for `foreign_key` with corresponding index(for posts and users table). I suggest you add that column and index in separate migration file. **When creating the reference column you should ensure that your `Like` belongs to an `:author`** and not to `user`. To help you just refer this **[article](https://stackoverflow.com/questions/27809342/rails-migration-add-reference-to-table-but-different-column-name-for-foreign-ke/42214988)**

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @djo1975 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230510132224_create_like.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 class CreateLike < ActiveRecord::Migration[7.0]
   def change
-    create_table :likes, &:timestamps
+    create_table :likes do |t|
+      t.references :posts, null: false, foreign_key: true
+
```
Comment (danmainah) [inline db/migrate/20230510132224_create_like.rb:6]: - [ ] @djo1975 great work implementing the changes requested however you forgot to add a foreign key reference column for the user to make the association between the user and like complete.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @djo1975 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Github flow was correctly used
✔️ Professional README file.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230510132224_create_like.rb (hunk 2 above)
Comment (djo1975) [reply db/migrate/20230510132224_create_like.rb:6]: Thanks, @danmainah.

**Review by brytebee — APPROVED** [review]
Hi, @djo1975 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by djo1975** [comment]
THX

**Pull request merged**